rpaas/acl.py

```
@@ -0,0 +1,148 @@
+        self.service_name = config.get("RPAAS_SERVICE_NAME", "rpaas")
+        self.acl_api_host = config.get("ACL_API_HOST")
+        self.acl_api_user = config.get("ACL_API_USER")
+        self.acl_api_password = config.get("ACL_API_PASSWORD")
+        self.acl_api_timeout = int(config.get("ACL_API_TIMEOUT", 30))
+        self.acl_port_range_start = config.get("ACL_PORT_RANGE_START", "32768")
+        self.acl_port_range_end = config.get("ACL_PORT_RANGE_END", "61000")
+        self.network_api_url = config.get("NETWORK_API_URL", None)
+        self.acl_auth_basic = HTTPBasicAuth(self.acl_api_user, self.acl_api_password)
+        if self.network_api_url:
+            self.network_api_username = config.get("NETWORK_API_USERNAME")
+            self.network_api_password = config.get("NETWORK_API_PASSWORD")
+            self.ip_client = Ip.Ip(self.network_api_url, self.network_api_username, self.network_api_password)
+            self.network_client = Network.Network(self.network_api_url, self.network_api_username,
+                                                  self.network_api_password)
+
+    def add_acl(self, name, src, dst):
+        src_network = self._get_network_from_ip(src)
+        if src_network == src:
+            src_network = str(ipaddress.ip_network(unicode(src_network)))
+        src = str(ipaddress.ip_network(unicode(src)))
+        dst = self._get_network_from_ip(dst)
+        if self._check_acl_exists(name, src, dst):
+            return
+        request_data = self._request_data("permit", name, src, dst)
+        response = self._make_request("PUT", "api/ipv4/acl/{}".format(src_network), request_data)
+        self._check_acl_response(response)
+        self.storage.store_acl_network(name, src, dst)
+
+    def remove_acl(self, name, src):
+        src = str(ipaddress.ip_network(unicode(src)))
+        acl_data = self.storage.find_acl_network({"name": name, "acls.source": src})
+        if acl_data and 'acls' in acl_data and acl_data['acls']:
```

You could fail early here to reduce indentation levels. Something like:
```
if not acl_data:
    return
acls = acl_data.get('acls')
if not acls:
    return
```